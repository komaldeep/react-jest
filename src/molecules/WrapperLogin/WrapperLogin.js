import React, { Component }  from 'react';
import PropTypes from 'prop-types';

import Button, {buttonSizes , buttonSkins} from './../../atoms/Button/Button';
import InputText from './../../atoms/InputText/InputText';
import Label, { labelOrigins } from './../../atoms/Label/Label'

export default class WrapperLogin extends Component {

  static displayName = 'molecule/WrapperLogin';

  static propTypes = {
    // notificationKey: PropTypes.oneOf(getEnumValues(loginNotifications)),
  };

  loginButton = () => {

  }

  render() {
    return (
      <div>
        <Label
          vertical={labelOrigins.VERTICAL}
        >
          Username
        </Label>
        <InputText
          type="text"
          placeholder="Enter your name"
          inputValue={this.inputFieldValue}
        />

        <Button
          size={buttonSizes.SM}
          skin={buttonSkins.SECONDARY}
          buttonClicked={this.loginButton}
        >
          Submit
        </Button>
      </div>
    );
  }
}
