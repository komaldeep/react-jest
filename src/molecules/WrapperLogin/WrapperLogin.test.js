import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import WrapperLogin from './WrapperLogin';

test('renders without crashing', () => {
  const rendered = renderer.create(<WrapperLogin />).toJSON();
  expect(rendered).toMatchSnapshot();
});

test('on click of button', () => {
  const component = mount( <WrapperLogin /> )
  const button = component.find('Button');
  button.simulate('click');
});