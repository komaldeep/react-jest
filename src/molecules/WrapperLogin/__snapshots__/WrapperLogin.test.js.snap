// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <label
    className="vertical"
  >
    Username
  </label>
  <div
    className="wrap"
  >
    <input
      className="input"
      disabled={false}
      id={undefined}
      maxLength={9999}
      name={undefined}
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value={undefined}
    />
  </div>
  <button
    className="btn small secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;
